### ISSUE TITLE
APK Demo - Tutorial

### ISSUE THREAD

**Issue by edwin-pixel-1** [GleasonK/android-webrtc-tutorial#4]
Hi, 
I'm developing an app to do broadcasting, however, Before to continous with my project I wanted to do a small test with the APK example made [here](http://kevingleason.me/AndroidRTC/get.html).

But I had some problems testing this app, I cannot get connections between two android devices, is always "Connecting..." but it works between one android device and [web](https://kevingleason.me/AndroidRTC/) api.

I would like to use this example like a base, to built a prototype to do video-broadcast.

I leave some specific questions in this [thread](https://github.com/GleasonK/android-webrtc-api/issues/3).
If someone help me with this, I would appreciate it.

Thanks!

**Comment by DatzZDeVesh** [comment]
have you resolved the issue

**Comment by PeterPan355** [comment]
I have the same problem..when I first installed the app, it connected devices but soon connection was closed. When I try to connect again, every time connection isn't established (only showing own front camera of device). Any suggestions?

**Comment by edwin-pixel-1** [comment]
Yes, I solved this issue. Currently I have an small prototype of a Video-Broadcasting app for android based on this example. I hope to publish an example of Video-Broadcasting the next week.

I had to make a test of this [template](https://github.com/GleasonK/android-webrtc-tutorial) using my own pubnub keys and everything worked well, obviously with some issues.

**Comment by PeterPan355** [comment]
Can you please help me with this problem? How did you solve it? Because I really don't know where the problem is..

**Comment by edwin-pixel-1** [comment]
Have you made everything of this example and compiled your own project from this template?
Are you usign your own pubnub keys?

**Comment by PeterPan355** [comment]
Yeah, created everything like there just in my project, checked a lot of times, but connection between devices can't be established, but works with web api..

**Comment by edwin-pixel-1** [comment]
@Skatic7 Check the permmisions, Have you changed `your.package.name`?
Have you published your code? Let me know where you have your code to check.

**Comment by ryandt** [comment]
Hey @alejo740. I'm having the same issue of being stuck at "Connecting..." on both devices (Android-to-Android) once a call has been answered. I am using my own demo pub key and sub key for this. Do you have a working prototype that you would be willing to share?

**Comment by PeterPan355** [comment]
@alejo740  @ryandt  Just one question, is it possible to establish connection between two androids with THIS tutorial? Or just between Android and web (just like it is example in tutorial)? Because you guys are saying that it stops at "connecting", but "connecting" isn't in this tutorial, but is in some other tutorials. :-/

**Comment by edwin-pixel-1** [comment]
Hello guys, currently I have working my App based on this tutorial. 
I had to do different things, but I suggest try the next:
- First off, I suggest you, donwload and put manually the 'PubNub Signaling Library'. See Image 1.
- Therefore, you have to remove the dependence: compile 'me.kevingleason:pnwebrtc:1.0.6@aar'

![image](https://cloud.githubusercontent.com/assets/929504/14517246/fe4c4286-01d0-11e6-8c92-16355df445f6.png)
**Image 1,**

**Comment by GlenGithub** [comment]
how to solve this 'connection' question. help me!!!

**Comment by devyaz** [comment]
Does this need a punhub server for signaling? Or can I use my own server?

**Comment by mkmlh** [comment]
@alejo740 have u tested between device? i've been trying so hard on this thing for a one month..and 0 result. Please help us

**Comment by guptasim8** [comment]
how can i get my own demo pub key and sub key ?

**Comment by mkmlh** [comment]
@guptasim8 sign up to pubnub

**Comment by WoogearKwon** [comment]
Please refer to this [link ](https://github.com/GleasonK/AndroidRTC/issues/22)below on the issue "connectin...".
Just download a new verison of tutorial including STUN/TURN servers matters from [here](https://github.com/newbie007fx/newwebrtc).
Additionally, just change the PUB_KEY & SUB_KEY in Constans.class with the ones from GleasonK's tutorial.
It will work!
